src/OddJobs/Job.hs

```diff
@@ -529,26 +529,26 @@ jobPoller = do
   traceM "jobPoller just before log statement"
   log LevelInfo $ LogText $ toS $ "Starting the job monitor via DB polling with processName=" <> processName
   concurrencyControlFn <- getConcurrencyControlFn
-  withResource pool $ \pollerDbConn -> forever $ concurrencyControlFn >>= \case
-    False -> do
+  withRunInIO $ \runInIO -> withResource pool $ \pollerDbConn -> forever $ runInIO concurrencyControlFn >>= \case
+    False -> runInIO $ do
       log LevelWarn $ LogText $ "NOT polling the job queue due to concurrency control"
       -- If we can't run any jobs ATM, relax and wait for resources to free up
       delayAction
     True -> do
       nextAction <- mask_ $ do
-        log LevelDebug $ LogText $ toS $ "[" <> processName <> "] Polling the job queue.."
-        r <- liftIO $ jobPollingIO pollerDbConn processName tname lockTimeout
+        runInIO $ log LevelDebug $ LogText $ toS $ "[" <> processName <> "] Polling the job queue.."
```

Comment:
is `runInIO` required here?

Reply:
I think for logging we need to be in the unlifted context (eg not IO)